src/test/java/online/talkandtravel/controller/http/integrationtest/ChatControllerIntegrationTest.java

```diff
@@ -28,27 +46,182 @@ class ChatControllerIntegrationTest extends IntegrationTest {
-
-    log.info("token {}", token);
-    // Act & Assert
-    mockMvc
-        .perform(
-            post("/api/chats/private")
-                .header(
-                    "Authorization",
-                    "Bearer " + token) // Add Authorization header with Bearer token
-                .contentType(MediaType.APPLICATION_JSON)
-                .content(objectMapper.writeValueAsString(newPrivateChatDto)))
-        .andExpect(status().isOk())
-        .andExpect(jsonPath("$").isNumber());
+  @Autowired private TestChatService testChatService;
+
+  private static final String TOKEN_PREFIX = "Bearer ";
+
+  private User alice, bob;
+
+  @BeforeEach
+  void init() {
+    alice = getAlice();
+    bob = getBob();
+  }
+
+  @Nested
+  class CreatePrivateChat {
+
+    private static final String CREATE_PRIVATE_CHAT_PATH = "/api/chats/private";
+
+    @Test
+    void createPrivateChat_shouldThrowConflict() throws Exception {
+      performChatCreationByAlice(bob,
+          status().isConflict());
+    }
+
+    @Test
+    void createPrivateChat_shouldReturnChatId_whenRequestIsValid() throws Exception {
+      performChatCreationByAlice(getTomas(),
+          status().isOk(),
+          jsonPath("$").isNumber());
+    }
+
+    @Test
+    void createPrivateChat_shouldThrow_whenTheSameUser() throws Exception {
+      performChatCreationByAlice(alice,
+          status().isBadRequest());
+    }
+
+    private void performChatCreationByAlice(User companion, ResultMatcher... matcher)
+        throws Exception {
+      NewPrivateChatDto content = new NewPrivateChatDto(companion.getId()); //companion
+      String aliceToken = authenticateUser(alice);
+
+      mockMvc.perform(
+              post(CREATE_PRIVATE_CHAT_PATH)
+                  .headers(getAuthHeader(aliceToken))
+                  .contentType(APPLICATION_JSON)
+                  .content(toJson(content)))
+          .andExpectAll(matcher);
+    }
+
+  }
+
+  @Nested
+  class FindMainChat {
+
+    private static final String FIND_MAIN_CHAT_PATH = "/api/v2/country/%s/main-chat";
+
+    private NewPrivateChatDto newPrivateChatDto;
+    private String token;
+
+    @BeforeEach
+    void init() {
+      newPrivateChatDto = new NewPrivateChatDto(alice.getId());
+      token = authenticateUser(alice);
+    }
+
+    @ParameterizedTest
+    @MethodSource("findMainChatArgs")
+    void findMainChat_shouldFindChat(String name, String id) throws Exception {
+      mockMvc.perform(
+              get(FIND_MAIN_CHAT_PATH.formatted(name))
+                  .headers(getAuthHeader(token))
+                  .contentType(APPLICATION_JSON)
+                  .content(toJson(newPrivateChatDto)))
+          .andExpect(status().isOk())
+          .andExpect(jsonPath("id").value(id))
+          .andExpect(jsonPath("name").value(name));
+    }
+
+    @ParameterizedTest
+    @MethodSource("findMainChat_shouldThrowArgs")
+    void findMainChat_shouldThrow(String countryName) throws Exception {
```

Comment:
I think we should rename this method. It doesn't throw exception. It just responses with message that 'conunty is not found'. Replace 'shouldThrow' with 'notFound' for example